reactor-core/src/test/java/reactor/core/publisher/MonoCacheTimeTest.java

```diff
@@ -900,4 +905,57 @@ public void scanSubscriber() {
 		assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
 	}
 
+	@Test
+    public void signalDependentWithSchedulerOperatorTests() {
+	    Duration valueTtl = Duration.ofMillis(1000);
+        Duration errorTtl = Duration.ofMillis(2000);
+        Duration emptyTtl = Duration.ofMillis(3000);
+
+        AtomicInteger subCount = new AtomicInteger();
+        Supplier<Mono<Integer>> valueMonoSupplier = () -> Mono.just(subCount.incrementAndGet());
+        Supplier<Mono<Integer>> errorMonoSupplier = () -> Mono.error(new RuntimeException("Boom #" + subCount.incrementAndGet()));
+        Supplier<Mono<Integer>> emptyMonoSupplier = () -> { subCount.incrementAndGet(); return Mono.empty(); };
+
+        VirtualTimeScheduler vts = VirtualTimeScheduler.create();
+        Function<Supplier<Mono<Integer>>, Mono<Integer>> monoFunction = monoSupplier -> Mono.defer(monoSupplier)
+                .cache(value -> valueTtl, error -> errorTtl, () -> emptyTtl, vts);
+
+        assertCacheTtl(monoFunction.apply(valueMonoSupplier), valueTtl, subCount, vts);
+        assertCacheTtl(monoFunction.apply(errorMonoSupplier), errorTtl, subCount, vts);
+        assertCacheTtl(monoFunction.apply(emptyMonoSupplier), emptyTtl, subCount, vts);
+    }
+
+    private void assertCacheTtl(Mono<Integer> mono, Duration ttl, AtomicInteger subCount, VirtualTimeScheduler vts) {
+	    subCount.set(0);
+
+        mono.subscribe();
```

Comment:
this leads to some log pollution in the case of the `errorMonoSupplier`, because no error handler is defined. To avoid that, better use `mono.subscribe(v -> {}, e -> {});` at the 3 spots where you used `mono.subscribe();`.

Reply:
Perfect,  I'll fix it